src/views/ModelPlan/TaskList/Basics/index.tsx

```diff
@@ -1,29 +1,439 @@
-import React from 'react';
-import { Route, Switch } from 'react-router-dom';
+import React, { Fragment, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link, Route, Switch, useHistory, useParams } from 'react-router-dom';
+import { useMutation, useQuery } from '@apollo/client';
+import {
+  Breadcrumb,
+  BreadcrumbBar,
+  BreadcrumbLink,
+  Button,
+  Dropdown,
+  IconArrowBack,
+  Label,
+  TextInput
+} from '@trussworks/react-uswds';
+import { Field, FieldArray, Form, Formik, FormikProps } from 'formik';
 
 import MainContent from 'components/MainContent';
-import NotFound, { NotFoundPartial } from 'views/NotFound';
+import PageHeading from 'components/PageHeading';
+import PageNumber from 'components/PageNumber';
+import Alert from 'components/shared/Alert';
+import AutoSave from 'components/shared/AutoSave';
+import CheckboxField from 'components/shared/CheckboxField';
+import { ErrorAlert, ErrorAlertMessage } from 'components/shared/ErrorAlert';
+import FieldErrorMsg from 'components/shared/FieldErrorMsg';
+import FieldGroup from 'components/shared/FieldGroup';
+import modelCategoryEnum from 'constants/enums/modelCategory';
+import GetModelPlanQuery from 'queries/GetModelPlanQuery';
+import {
+  GetModelPlan,
+  GetModelPlanVariables
+} from 'queries/types/GetModelPlan';
+import { UpdateModelPlan as UpdateModelPlanType } from 'queries/types/UpdateModelPlan';
+import UpdateModelPlan from 'queries/UpdateModelPlan';
+import flattenErrors from 'utils/flattenErrors';
+import {
+  translateCmmiGroup,
+  translateCmsCenter,
+  translateModelCategory
+} from 'utils/modelPlan';
+import planBasicsSchema from 'validations/planBasics';
+import { NotFoundPartial } from 'views/NotFound';
+
+import Milestones from './Milestones';
+import Overview from './Overview';
+
+type PlanBasicModelPlanFormType = {
+  modelName: string;
+  modelCategory: string | null;
+  cmsCenters: string[];
+  cmmiGroup: string[];
+  cmsOther: string;
+};
+
+const BasicsContent = () => {
+  const { t } = useTranslation('basics');
+  const { t: h } = useTranslation('draftModelPlan');
+  const { modelId } = useParams<{ modelId: string }>();
+
+  const formikRef = useRef<FormikProps<PlanBasicModelPlanFormType>>(null);
+  const history = useHistory();
+  const [isCmmiGroupShown, setIsCmmiGroupShown] = useState(false);
+
+  const { data } = useQuery<GetModelPlan, GetModelPlanVariables>(
+    GetModelPlanQuery,
+    {
+      variables: {
+        id: modelId
+      }
+    }
+  );
+
+  const { modelName, modelCategory, cmsCenters, cmmiGroups, cmsOther } =
+    data?.modelPlan || {};
+
+  const [update] = useMutation<UpdateModelPlanType>(UpdateModelPlan);
+
+  const handleFormSubmit = (
```

Comment:
We need to ensure a user can't erase a model name.  Because autosave and `save and exit` don't use the form validation to progress, we need another check to ensure those two saves can't happen if the model name is empty.  Currently I can erase the model name, autosave, and it is permanently erased.  This may need to happen in the submit handlers to ensure no saves can happen in that field is empty.

Reply:
It might be worth considering what pages get auto-saved, if possible. Maybe certain ones (like this whole page, since it has no long-form text input) don't even have auto-save, just to simplify things?

Given that the inputs on this page are
- Model Name (do we want that to be auto-saved?)
- Category (Dropdown)
- Centers (Multi-select/dropdown)
- CMMI Groups (Multi-select/dropdown)
- "CMSOther" (A small text field)

maybe we just say "eh, this page doesn't do auto-save"

Is that taking it too far? Thoughts?

Reply:
I think the `modelName` is the only field on the entire form that we don't want to autosave.  Everything else, even if required by validation is fine, as they won't be able to progress without filling it out.  I can also see the side of just removing autosave if it isn't providing value on forms with ~2 input